### PR TITLE
feature/add-monitoring-disable-plan-for-prometheus

### PR DESCRIPTION
**Pull request by jschoewe**
Added bolt plan to enable/disable monitoring in prometheus by changing a puppet fact on the node. Also added script to refresh Prometheus targets from puppetdb with the matching fact and prometheus resource

### CODE REVIEW

File: tasks/monitoring_prometheus.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import os
+import sys
+import requests
+import socket
+import datetime
+import time
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..', 'python_task_helper', 'files'))
+from task_helper import TaskHelper
+
+class PrometheusAlertSilence(TaskHelper):
```
Comment (nmaludy) [inline tasks/monitoring_prometheus.py:24]: This should be written in Ruby like the other HTTP code is

File: plans/monitoring_solarwinds_prometheus.pp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#               ENC[PKCS7,xxx]
+#       targets:
+#         - prometheus.domain.tld
+#
+plan patching::monitoring_solarwinds_prometheus (
+  TargetSpec                $targets,
+  Enum['enable', 'disable'] $action,
+  Boolean                   $noop = false,
+) {
+  run_plan('patching::monitoring_solarwinds', $targets,
```
Comment (nmaludy) [inline plans/monitoring_solarwinds_prometheus.pp:51]: I expected a plan that is generic like we talked about, not hard coded for SolarWinds + Prometheus.

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: metadata.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "encore-patching",
-  "version": "1.1.1",
+  "version": "1.2.1",
```
Comment (nmaludy) [inline metadata.json:3]: should be version 1.2.0, right?

File: plans/monitoring_multiple.pp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+#             _plugin: pkcs7
+#             encrypted_value: >
+#               ENC[PKCS7,xxx]
+#       targets:
+#         - prometheus.domain.tld
+#
+plan patching::monitoring_multiple (
+  TargetSpec                $targets,
+  Enum['enable', 'disable'] $action,
+  Array                     $monitoring_plan = get_targets($targets)[0].vars['patching_monitoring_plan'],
```
Comment (nmaludy) [inline plans/monitoring_multiple.pp:50]: you're going to need to use a different key here

File: plans/monitoring_multiple.pp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  Boolean                   $noop = false,
+) {
+
+  # Loop over and run each monitoring plan
+  $monitoring_plan.each |String $plan| {
+    run_plan($plan, $targets,
+              action => $action,
+              noop   => $noop)
+  }
+}
```
Comment (nmaludy) [inline plans/monitoring_multiple.pp:60]: missing newline at end of the file

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: plans/monitoring_multiple.pp (hunk 4 above)
Comment (nmaludy) [reply plans/monitoring_multiple.pp:50]: if you use `patching_monitoring_plan` as an array, it's going to fail when you call `bolt plan run patching`.

Instead this key should be something like `patching_monitoring_plan_multiple` or similar.

Your vars would then look like:
```
vars:
  patching_monitoring_plan: 'patching::monitoring_multiple'
  patching_monitoring_plan_multiple:
    - 'patching::monitoring_solarwinds'
    - 'patching::monitoring_prometheus'
 ```

**Review by nmaludy — APPROVED** [review]

**Comment by nmaludy** [comment]
@jschoewe i think you just need to get build to pass

**Comment by bishopbm1** [comment]
@nmaludy can this be merged?

**Pull request merged**